src/utils/utils_hetero.py

```diff
@@ -86,36 +138,40 @@ def create_hetero_data_instance(graph, y_bus):
-    edge_attr = graph.line[['r_ohm_per_km',
-                           'x_ohm_per_km',
-                           'c_nf_per_km',
-                           'g_us_per_km',
-                           'max_i_ka',
-                           'parallel',
-                           'df',
-                           'length_km']].reset_index(drop=True)
+    # edge_index = pd.concat([edge_index, swapped], axis=0).reset_index(drop=True) should we include this line???
+
+    edge_attr_lines_list = []
+    for edges in graph.line.itertuples():
+        # Calculate line admittance from impedance and convert to per-unit system
+        r_tot = float(edges.r_ohm_per_km) * float(edges.length_km)
+        x_tot = float(edges.x_ohm_per_km) * float(edges.length_km)
+        conductance_line, susceptance_line = impedance_to_admittance(r_tot, x_tot, graph.bus['vn_kv'][edges.from_bus], graph.sn_mva)
+        edge_attr_lines_list.append([conductance_line, susceptance_line])
+    
+    # we are missing the swapped edges attributes, to add, create another for loop and uncomment the line above
+    edge_attr = pd.DataFrame(edge_attr_lines_list, columns=['r_tot', 'x_tot'])
     
+
+
     # Get edges for transformers, reindex, and make bidirectional
     edge_index_trafo = graph.trafo[['lv_bus', 'hv_bus']].reset_index(drop=True)
     edge_index_trafo['lv_bus'] = edge_index_trafo['lv_bus'].map(index_mapping)
     edge_index_trafo['hv_bus'] = edge_index_trafo['hv_bus'].map(index_mapping)
     swapped_trafo = deepcopy(edge_index_trafo)
     swapped_trafo[['lv_bus', 'hv_bus']] = edge_index_trafo[['hv_bus', 'lv_bus']]
+    # edge_index_trafo = pd.concat([edge_index_trafo, swapped_trafo], axis=0).reset_index(drop=True) should we include this line???
     
     
     # create a edge attribute dataframe for the transformers
-    edge_attr_trafo = pd.DataFrame(columns=['one', 'two', 'three']) # what are the computed features called?
-    edge_attr_list = []
+    edge_attr_trafo_list = []
     for trafo in graph.trafo.itertuples():
-        # Calculate transformer attributes
-        vkr_pu = float(trafo.vkr_percent / (trafo.sn_mva / (trafo.vn_lv_kv * math.sqrt(3))))
-        p_pu = float(math.sqrt((trafo.vk_percent ** 2) - (trafo.vkr_percent) ** 2) / (trafo.sn_mva / (trafo.vn_lv_kv * math.sqrt(3))))
-        q_pu = float(1.0)
-        edge_attr_list.append([vkr_pu, p_pu, q_pu])
+        r_tot = 0.0
+        x_tot = (trafo.vk_percent / 100) * (trafo.vn_lv_kv ** 2) / trafo.sn_mva
+        conductance, susceptance = impedance_to_admittance(r_tot, x_tot, trafo.vn_lv_kv, graph.sn_mva)
+        edge_attr_trafo_list.append([conductance, susceptance])
     
-    edge_attr_trafo = pd.DataFrame(edge_attr_list, columns=['one', 'two', 'three'])
+    # we are missing the swapped edges attributes, to add, create another for loop and uncomment the line above
+    edge_attr_trafo = pd.DataFrame(edge_attr_trafo_list, columns=['conductance', 'susceptance'])
```

Comment:
Same holds here as for the lines, the attributes are the same in both directions, so I don't think adding swapped attributes is necessary.